src/client/render/secondstage.cpp

```diff
@@ -238,3 +277,14 @@ RenderStep *addPostProcessing(RenderPipeline *pipeline, RenderStep *previousStep
 
 	return effect;
 }
+
+void ResolveMSAAStep::run(PipelineContext &context)
+{
+	u32 msaa_fbo_id = msaa_fbo->getGLBufferID(context);
+	u32 target_fbo_id = target_fbo->getGLBufferID(context);
+	GL.BindFramebuffer(GL.READ_FRAMEBUFFER, msaa_fbo_id);
+	GL.BindFramebuffer(GL.DRAW_FRAMEBUFFER, target_fbo_id);
+	v2u32 size = context.target_size;
+	GL.BlitFramebuffer(0, 0, size.X, size.Y, 0, 0, size.X, size.Y,
+			GL.COLOR_BUFFER_BIT | GL.DEPTH_BUFFER_BIT | GL.STENCIL_BUFFER_BIT, GL.NEAREST);
```

Comment:
Gave it a try. Advantages:

- Somewhat fits into the existing architecture instead of weirdly working around it

Disadvantages:

- It's harder to follow and understand the code because of the interface stuff
- Code duplication between old and new driver
- More code, even without duplication
- When doing this inline, I could be sure that GL version requirements would be met, since I knew the code would only run on GL >= 3.2 (where multisample textures are supported) anyway. Now I can't be sure and have to do extra GL version checks.